java-frontend/src/main/java/org/sonar/java/resolve/TypeVariableJavaType.java

```diff
@@ -53,7 +53,7 @@ public boolean isSubtypeOf(Type superType) {
       return true;
     }
     for (JavaType bound : bounds()) {
-      if (bound.isSubtypeOf(supType)) {
+      if (bound.isSubtypeOf(supType) || (supType.isParameterized() && bound == supType.erasure())) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)